frontend/src/contexts/wishlist-context.tsx

```diff
@@ -0,0 +1,79 @@
+import { createContext, ReactNode, useState } from 'react';
+
+import { WishlistInput } from '@autoline/shared/common/types/types';
+import { HeartIcon } from '@components/common/icons/icons';
+import { Notification } from '@components/common/notification/notification';
+import {
+  useCreateWishlistMutation,
+  useDeleteWishlistMutation,
+  useGetWishlistStatusQuery,
+} from '@store/queries/preferences/wishlist';
+
+type WishlistContextType = {
+  likedCars: string[] | undefined;
+  likeClick: (data: WishlistInput) => void;
+};
+
+const WishlistContext = createContext<WishlistContextType>({
+  likedCars: undefined,
+  likeClick: () => undefined,
+});
+
+const WishlistContextProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [isMessageHidden, setIsMessageHidden] = useState<boolean>(true);
```

Comment:
It's usually easier to comprehend "straight" statements: `open`, `show`, `render` than their negated analogs like `hide`.
So I would rather use the `isMessageOpen` name here.

The same can be said for the `if` statements. It's usually better to write `if(variable === 'value') {do smth} else {do else}` than `if(variable !== 'value') {do smth} else {do else}`